fix: do not send IPv6 DNS queries when IPv6 is disabled on the host

Author: Jan Vcelak <jvcelak@redhat.com>
Resolves: #835012
Upstream ITS: #7326

diff --git a/libraries/libldap/os-ip.c b/libraries/libldap/os-ip.c
index 2864256..c3e0876 100644
--- a/libraries/libldap/os-ip.c
+++ b/libraries/libldap/os-ip.c
@@ -584,8 +584,7 @@ ldap_connect_to_host(LDAP *ld, Sockbuf *sb,
 
 #if defined( HAVE_GETADDRINFO ) && defined( HAVE_INET_NTOP )
 	memset( &hints, '\0', sizeof(hints) );
-#ifdef USE_AI_ADDRCONFIG /* FIXME: configure test needed */
-	/* Use AI_ADDRCONFIG only on systems where its known to be needed. */
+#ifdef AI_ADDRCONFIG
 	hints.ai_flags = AI_ADDRCONFIG;
 #endif
 	hints.ai_family = ldap_int_inet4or6;
-- 
1.7.11.4

